File: examples/python/custom-model-training/dataset/asticaDataset_sample_add.py
import os
import requests
import json
import base64
import random

astica_api_key = 'API KEY HERE';    #visit https://astica.ai
astica_model_id = 10;               #model_id to update
astica_class_title = 'Dog'          #class will be created if it does not exist.
astica_upload_path = 'image.jpg'    # File path or Folder path (PNG or JPEG)


def asticaTrain_directory_upload(model_id, class_title, directory_path):
    for root, _, files in os.walk(directory_path):
        for file in files:
            file_path = os.path.join(root, file)
            asticaTrain_file_prepare(file_path)
            
def asticaTrain_file_upload(model_id, class_title,file_path):
    with open(file_path, 'rb') as file:
        encoded_file = base64.b64encode(file.read()).decode('utf-8')            
    #You can refer to uploaded samples this way.
    sample_uid = "{}_{}".format(os.path.basename(file_path), str(random.randint(1, 100000)))    
    #Send training data to api
    asticaTrain_upload(model_id, class_title,encoded_file,sample_uid)

def asticaTrain_upload(model_id, class_title, encoded_file, sample_uid):
    global astica_api_key
    astica_endpoint = 'https://train.astica.ai/model/upload'
    astica_requestData = {
        'tkn': astica_api_key,
        'model_id': model_id,
        'model_class': class_title,
        'input': encoded_file,
        'uid': sample_uid,
    }    
    try:
        response = requests.post(astica_endpoint, data=json.dumps(astica_requestData), headers={'Content-Type': 'application/json'})
        response.raise_for_status()  # Raise an exception if the request was unsuccessful

        data = json.loads(response.text)
        print(data)

        if data['status'] != 'OK':
            print("api request failed")
            print(data["error"])
        else:
            # Handle the success response
            print("Upload successful")

    except requests.exceptions.RequestException as e:
        print("Error:", str(e))

if os.path.isdir(astica_upload_path):
    asticaTrain_directory_upload(astica_model_id, astica_class_title, astica_upload_path)
else:
    asticaTrain_file_upload(astica_model_id, astica_class_title, astica_upload_path)
   
    

File: examples/node/custom-model-training/dataset/asticaDataset_sample_add.js
const axios = require('axios');
const fs = require('fs');
const path = require('path');
const Base64 = require('js-base64').Base64;

let astica_api_key = 'API KEY HERE';    //visit https://astica.ai
let astica_model_id = 0;               //model_id to update
let astica_class_title = 'Dog'          //class will be created if it does not exist.
let astica_upload_path = 'image.jpg'    // File path or Folder path (PNG or JPEG)

/* 
    You can retrieve your model_id:
      Using astica dashboard: https://astica.ai/account/models/ 
      Using API: https://train.astica.ai/dataset/list
*/


async function asticaTrain_directory_upload(model_id, class_title, directory_path){
    fs.readdir(directory_path, (err, files) => {
        files.forEach(file => {
            let filePath = path.join(directory_path, file);
            asticaTrain_file_upload(model_id, class_title, filePath)
        });
    });
}

async function asticaTrain_file_upload(model_id, class_title, filePath){
    let file = fs.readFileSync(filePath, { encoding: 'base64' });
    let sample_uid = path.basename(filePath) + "_" + Math.floor(Math.random() * 100000);
    await asticaTrain_upload(model_id, class_title, file, sample_uid)
}

async function asticaTrain_upload(model_id, class_title, encoded_file, sample_uid){
    let astica_endpoint = 'https://train.astica.ai/model/upload';
    let astica_requestData = {
        tkn: astica_api_key,
        model_id: model_id,
        model_class: class_title,
        input: encoded_file,
        uid: sample_uid,
    }

    try{
        let response = await axios.post(astica_endpoint, astica_requestData, { headers: { 'Content-Type': 'application/json'}})
        
        if(response.data.status != 'OK'){
            console.log("api request failed");
            console.log(response.data.error);
        } else {
            // Handle the success response
            console.log("Upload successful");
        }

    }catch(error){
        console.log("Error: ", error);
    }
}

if(fs.lstatSync(astica_upload_path).isDirectory()){
    asticaTrain_directory_upload(astica_model_id, astica_class_title, astica_upload_path);
}else{
    asticaTrain_file_upload(astica_model_id, astica_class_title, astica_upload_path);
}